src/main/groovy/lang/ObjectRange.java

```diff
@@ -488,37 +500,61 @@ public void remove() {
 
     @Override
     public List step(int step) {
+        return step((Number) step);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List step(Number step) {
         final IteratorClosureAdapter adapter = new IteratorClosureAdapter(this);
         step(step, adapter);
         return adapter.asList();
     }
 
     /**
-     * Increments by one
+     * Increments by step size
      *
      * @param value the value to increment
-     * @return the incremented value
+     * @param step  the value to increment by
+     * @return the incremented value or null, if there isn't any
      */
-    private static Object increment(Object value) {
+    protected static Comparable increment(Comparable value, Number step) {
         if (value instanceof Number) {
-            return plus((Number) value, 1);
-        } else {
-            return InvokerHelper.invokeMethod(value, "next", null);
+            return (Comparable) plus((Number) value, step);
+        }
+
+        for (int i = 0; compareLessThan(i, step); i++) {
```

Comment:
I consider throwing an IllegalArgumentException in ObjectRange for non-Integer steps when the starting value is not a number an alternative. Same for IntRange, it may be argued that an IntRange should not be steppable by anything else than an int, i.e. clients may expect the output of stepping an IntRange to be ints only.

Overall, I think I might generally prefer to rename the method step() to something else for non-int "step" values, because it does not step through the values the range represents, but instead it creates a new range with different values. something like scale() or toNewRange().